src/main/java/net/dv8tion/jda/api/entities/ChannelField.java

```diff
@@ -21,35 +21,175 @@
 import javax.annotation.Nonnull;
 import javax.annotation.Nullable;
 
-//TODO-v5: Needs docs
+
+/**
+ * This enum represents the attributes of a channel that can be modified by events.
+ * <br><b>Most</b> of these changes are tracked and reflected by {@link AuditLogKey Audit Log Entries}.
+ *
+ * @see net.dv8tion.jda.api.events.channel.GenericChannelEvent
+ * @see AuditLogKey
+ */
 public enum ChannelField
 {
     //TODO-v5: Should these be the REST JSON names (camelCase), the AuditLogKey names (snake_case), or JDA's generic naming (Event.IDENTIFIER)
     //TODO-v5: Current using JDA's generic namings
 
-    //TODO-v5: All of these need explicit docs
+    //Generic
+
+    /**
+     * The {@link ChannelType type} of the channel.
+     *
+     * @see Channel#getType()
+     */
     TYPE("type", AuditLogKey.CHANNEL_TYPE),
+
+    /**
+     * The name of the channel.
+     *
+     * @see Channel#getName()
+     */
     NAME("name", AuditLogKey.CHANNEL_NAME),
 
+    /**
+     * The {@link Category parent} of the channel.
+     *
+     * Limited to {@link BaseGuildMessageChannel Base Guild Channels} (and implementations).
+     *
+     * @see BaseGuildMessageChannel#getParentCategory()
+     */
+    PARENT("parent", AuditLogKey.CHANNEL_PARENT),
+
+    /**
+     * The position of this channel relative to other channels in the guild.
+     *
+     * @see IPositionableChannel#getPosition()
+     * //todo-v5 if the calculation of IPositionableChannel#getPosition() changes, this may need modification too.
+     */
     POSITION("position", null), //Discord doesn't track Channel position changes in AuditLog.
 
+
+    //Text Specific
+
+    /**
+     * The topic of the channel.
+     *
+     * Limited to {@link BaseGuildMessageChannel Base Guild Channels} (and implementations).
+     *
+     * @see BaseGuildMessageChannel#getTopic()
+     */
     TOPIC("topic", AuditLogKey.CHANNEL_TOPIC),
+
+    /**
+     * The NSFW state of the channel.
+     *
+     * Limited to {@link BaseGuildMessageChannel Base Guild Channels} (and implementations).
+     *
+     * @see BaseGuildMessageChannel#isNSFW()
+     */
     NSFW("nsfw", AuditLogKey.CHANNEL_NSFW),
+
+    /**
+     * The state of slow mode in the channel.  This defines the minimum time between message sends.
+     *
+     * Limited to {@link TextChannel Text Channels}.
+     *
+     * @see TextChannel#getSlowmode()
+     */
     SLOWMODE("slowmode", AuditLogKey.CHANNEL_SLOWMODE),
 
-    PARENT("parent", AuditLogKey.CHANNEL_PARENT),
 
+    //Voice Specific
+
+    /**
+     * The bitrate (in bits per second) of the audio in this channel.
+     *
+     * For standard channels this is between 8000 and 96000.
+     *
+     * VIP servers extend this limit to 128000.
```

Comment:
Are these values stored anywhere in JDA? if they are we should use `@value`

Reply:
Kind of - there's a Guild.BoostTier enum, but its values aren't easily accessible to link in the javadoc.  I have linked that enum in the docs now, though.